lasso/emme.py

```diff
@@ -793,14 +797,266 @@ def prepare_table_for_taz_transit_network(
         links_df = pd.concat([links_df, new_link_gdf], sort = False, ignore_index = True)
         
         links_df.drop_duplicates(subset = ['A', 'B'], inplace = True)
+
+    ### PNR edits -- create PNR dummy links, pnr/knr/walk connectors, transfer links
+
+    ### 1. create PNR dummy links
+    # read pnr parking location
+    pnr_nodes_df = pd.read_csv(parameters.pnr_node_location)[['Zone','Station_Type','X','Y','Vehicle_Cap','Headway']]
+    pnr_nodes_df = gpd.GeoDataFrame(
+        pnr_nodes_df, 
+        geometry=gpd.points_from_xy(pnr_nodes_df['X'], pnr_nodes_df['Y']),
+        crs=parameters.output_proj)
+    pnr_nodes_df = pnr_nodes_df.to_crs(parameters.output_proj)
+    pnr_nodes_df["X"] = pnr_nodes_df["geometry"].apply(lambda g: g.x)
+    pnr_nodes_df["Y"] = pnr_nodes_df["geometry"].apply(lambda g: g.y)
+
+    # reformat pnr_nodes_df, add missing columns
+    for c in nodes_df.columns:
+        if c not in pnr_nodes_df.columns:
+            if c not in ['county']:
+                pnr_nodes_df[c] = 0
```

Comment:
Is it safe to assign all 0? Double check if any node attributes are used in tm2py.